src/Util/Message.cc

```diff
@@ -107,13 +109,34 @@ CARTA::AddRequiredTiles Message::AddRequiredTiles(
     return add_required_tiles;
 }
 
-CARTA::Point Message::Point(int x, int y) {
+CARTA::Point Message::Point(float x, float y) {
     CARTA::Point point;
     point.set_x(x);
     point.set_y(y);
     return point;
 }
 
+CARTA::Point Message::Point(const casacore::Vector<casacore::Double>& input, int x_index, int y_index) {
+    CARTA::Point point;
+    point.set_x((float)input(x_index));
+    point.set_y((float)input(y_index));
+    return point;
+}
+
+CARTA::Point Message::Point(const std::vector<casacore::Quantity>& input, int x_index, int y_index) {
+    CARTA::Point point;
+    point.set_x((float)input[x_index].getValue());
+    point.set_y((float)input[y_index].getValue());
+    return point;
+}
+
+CARTA::Point Message::Point(const std::vector<double>& input, int x_index, int y_index) {
+    CARTA::Point point;
+    point.set_x((float)input[x_index]);
+    point.set_y((float)input[y_index]);
+    return point;
+}
+
```

Comment:
These are still duplicating the internals of `Message::Point(float x, float y)`. You should be *calling* that function from inside these functions instead (and I think that if you do that you won't need the explicit casts).